src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourceMappingOptionsControl.xaml.cs

```diff
@@ -0,0 +1,250 @@
+        {
+            SourceMappingsCollection.Remove((MappingUIDisplay)packageList.SelectedItem);
+            (ClearButtonCommand as ButtonCommand).InvokeCanExecuteChanged();
+        }
+
+        private bool CanExecuteRemoveButtonCommand(object parameter)
+        {
+            return SourceMappingsCollection.Count > 0;
+        }
+
+        private void ExecuteClearButtonCommand(object parameter)
+        {
+            SourceMappingsCollection.Clear();
+            (RemoveButtonCommand as ButtonCommand).InvokeCanExecuteChanged();
+        }
+
+        private bool CanExecuteClearButtonCommand(object parameter)
+        {
+            return SourceMappingsCollection.Count > 0;
+        }
+
+        internal bool ApplyChangedSettings()
+        {
+            IReadOnlyDictionary<string, IReadOnlyList<string>> Patterns = new Dictionary<string, IReadOnlyList<string>>();
+            PackageSourceMapping packageSourceMappings = new PackageSourceMapping(Patterns);
+            var componentModel = NuGetUIThreadHelper.JoinableTaskFactory.Run(ServiceLocator.GetComponentModelAsync);
+            var settings = componentModel.GetService<ISettings>();
+            PackageSourceMappingProvider packageSourceMappingProvider = new PackageSourceMappingProvider(settings);
+            IReadOnlyList<PackageSourceMappingSourceItem> packageSourceMappingsSourceItems = ReadMappingsFromUIToConfig(SourceMappingsCollection);
+            try
+            {
+                if (SourceMappingsChanged(_originalPackageSourceMappings, packageSourceMappingsSourceItems))
+                {
+                    packageSourceMappingProvider.SavePackageSourceMappings(packageSourceMappingsSourceItems);
+                }
+            }
+            // Thrown during creating or saving NuGet.Config.
+            catch (NuGetConfigurationException ex)
+            {
+                MessageHelper.ShowErrorMessage(ex.Message, Resx.ErrorDialogBoxTitle);
+                return false;
+            }
+            // Thrown if no nuget.config found.
+            catch (InvalidOperationException ex)
+            {
+                MessageHelper.ShowErrorMessage(ex.Message, Resx.ErrorDialogBoxTitle);
+                return false;
+            }
+            catch (UnauthorizedAccessException)
+            {
+                MessageHelper.ShowErrorMessage(Resx.ShowError_ConfigUnauthorizedAccess, Resx.ErrorDialogBoxTitle);
+                return false;
+            }
+            // Unknown exception.
+            catch (Exception ex)
+            {
+                MessageHelper.ShowErrorMessage(Resx.ShowError_ApplySettingFailed, Resx.ErrorDialogBoxTitle);
+                ActivityLog.LogError(NuGetUI.LogEntrySource, ex.ToString());
+                return false;
+            }
+            return true;
+        }
+
+        // Returns true if there are no changes between existingSourceMappings and packageSourceMappings.
```

Comment:
Is opposite true? I can see if there is change then it's returning `true`.

Reply:
Yes. I accidentally wrote the opposite of what the method does